wip: Навыки

vovakons opened the pull request:
## Что этот ПР делает

Добавление навыков.
https://docs.google.com/spreadsheets/d/1dIYQE7mP2_Lu0nQVUoc4T6mu2EGVO37EAP49geGnXuw/edit?usp=sharing
## Почему это хорошо для игры

Хорошо ведь?

## Демонстрация изменений

<details><summary>Демонстрации изменений</summary>
<p>

Тут будут скриншоты и видео, но потом.

</p>
</details>


## Тестирование

Сейчас в работе.

vovakons committed:
skills basic datums

Датумы для навыков.

vovakons committed:
basic architecture of skills system

Базовая архитектура системы скилов.
Работа через сигналы.
Получение уровня навыка в зависимости от профессии.
Навык перетаскивания.

vovakons committed:
engineering skills

Реализация навыков контруирования и строительства.
Пустые шаблоны для некоторых скиллов.

littleboobs commented:
Сделай, пожалуйста, нормально, портанув с ТГ, а не изобретай велосипед, потому что тебе просто прикольно. Система существующая годами, портированная на несколько проектов, разросшаяся и проверенная, обросшая багфиксами, юнит тестами и т.д. + расширяемость при портированиях и новоделах от шизов лунопопиков. Ничего из этого в твоей системе, которую ты на коленке изобретаешь, даже поглядывая на референс, не будет. 🙏

vovakons committed:
Merge branch 'master220' into skills

PiroMage commented:
Будто бы взять аналог с ТГ, или с СМов будет лучше, там уже по сути все готовое и остается только убрать для антагов требование навыков (ну или дать им макс навыки).

PlayerUnknown14 commented:
Вова, делай ТГ навыки 🙏

vovakons commented:
> Вова, делай ТГ навыки 🙏

открой тг навыки прежде чем советовать такое.

vovakons committed:
Merge branch 'master220' into skills

vovakons committed:
refactoring and clean code

Подчистил код, перенес код скиллов в модуль.
Удалил датумы уровней навыков, сделал через обычный int и списки
модификаторов.

vovakons committed:
fix merge issue

tsoivecen199015 commented:
Даже Иван говорит что велосипед изобретать не нужно. И создание системы навыков это непосильная задача для одного непросвещённого человека. У нас есть всякие FATE, GURPS, D&D, люди думали десятилетиями над балансом, эти системы просты, большая часть из них может быть выражена в процентах. Но работа ещё в процессе, так что может быть всё изменится (и оно должно измениться).

vovakons commented:
> Даже Иван говорит что велосипед изобретать не нужно. И создание системы навыков это непосильная задача для одного человека.

<img width="410" height="228" alt="image" src="https://github.com/user-attachments/assets/a5ab127f-2347-490a-9734-5fd9e76d41f4" />

vovakons committed:
engineering building skill

Доделан навык строительства до конца.
Сигналы навыков перенесены в новый файл.
Создана мапа для хранения уровня навыка через mind.

vovakons committed:
not create new skill datums (reuse one)

Теперь уровни навыков хранятся в мапе в самом майнде.
Датумы навыков отвязаны от персонажей, получают уровень навыка через
сигнал, а сами датумы переиспользуются всеми персонажами (stateless).

vovakons committed:
construction skill

Реализация навыка конструирования до конца.
Частично реализован навык взлома.

vovakons committed:
add all skills datums

Добавлены все датумы навыков.

vovakons committed:
skills window

Добавлено tgui окно для показа текущих навыков персонажа.
Открывается через IC -> Навыки персонажа.

vovakons committed:
electicity skill

Реализован навык электрики.
Влияет на шанс ударом током.
Влияет на показ проводки:
- без навыка проводка будет каждый раз перемешиваться.
- с начинающим навыком и ниже не будет показана информация.

vovakons committed:
setup engineering jobs skills

Выставлены уровни навыков для инженерных ролей.

vovakons committed:
skill macros fixes

vovakons committed:
atmos skill

vovakons committed:
Merge branch 'master220' into skills

vovakons committed:
Merge branch 'master220' into skills

vovakons committed:
general skills and remove cargo skills

Удален радел навыков карго, создан раздел общих навыков.

vovakons committed:
Merge branch 'master220' into skills

vovakons committed:
fix after merge

shahribek committed:
climbing into mech and dirivng mech skills

shahribek committed:
mecha power usage

vovakons committed:
use alist in skills

Использование ассоциативного листа в скиллах для использования int как
ключей.

shahribek committed:
bug fix

shahribek committed:
bug fix & mod_suit skill modifiers

shahribek committed:
another bug fix

vovakons committed:
accuracy skill

Реализован навык точности, влияет на разброс и шанс попадания.
Расставлены навыки для ролей отдела СБ.

vovakons committed:
Merge branch 'skills' of github.com:vovakons/Paradise into skills

vovakons committed:
unavailable skill level bug fixed

shahribek committed:
mind_skill_initialization

shahribek committed:
Grep Check issues fix 1

shahribek committed:
bug fix
fixed bug with projectile.firer being null when excepted to not be

shahribek committed:
codestyle fix

shahribek committed:
Revert "codestyle fix" and keep wanted changes

This reverts commit 579d9aaa121d0d41b59aa3ab0a6635d6eba1f60d.
and keep changes in SkillsWin.tsx file

shahribek committed:
tgui bundle update

shahribek committed:
spacepod battery usage depended on skill
